python/paddle/fluid/tests/unittests/test_lerp_op.py

```diff
@@ -0,0 +1,146 @@
+        self.shape = [2, 2, 25]
+
+
+class TestLerpWithDim4(TestLerp):
+    def init_shape(self):
+        self.shape = [2, 2, 5, 5]
+
+
+class TestLerpWithDim5(TestLerp):
+    def init_shape(self):
+        self.shape = [2, 1, 2, 5, 5]
+
+
+class TestLerpWithDim6(TestLerp):
+    def init_shape(self):
+        self.shape = [2, 1, 2, 5, 1, 5]
+
+
+class TestLerpAPI(unittest.TestCase):
+    def init_dtype(self):
+        self.dtype = 'float32'
+
+    def setUp(self):
+        self.init_dtype()
+        self.x = np.arange(1., 5.).astype(self.dtype)
+        self.y = np.full(4, 10.).astype(self.dtype)
+        self.w = np.asarray([0.5]).astype(self.dtype)
+        self.res_ref = self.x + 0.5 * (self.y - self.x)
+        self.place = [paddle.CPUPlace()]
+        if core.is_compiled_with_cuda():
+            self.place.append(paddle.CUDAPlace(0))
+
+    def test_static_api(self):
+        paddle.enable_static()
+
+        def run(place):
+            with paddle.static.program_guard(paddle.static.Program()):
+                x = paddle.fluid.data('x', [1, 4], dtype=self.dtype)
+                y = paddle.fluid.data('y', [1, 4], dtype=self.dtype)
+                w = paddle.fluid.data('w', [1], dtype=self.dtype)
+                out = paddle.lerp(x, y, w)
+                exe = paddle.static.Executor(place)
+                res = exe.run(feed={
+                    'x': self.x.reshape([1, 4]),
+                    'y': self.y.reshape([1, 4]),
+                    'w': self.w
+                })
+            for r in res:
+                self.assertEqual(np.allclose(self.res_ref, r), True)
+
+        for place in self.place:
+            run(place)
+
+    def test_dygraph_api(self):
+        def run(place):
+            paddle.disable_static(place)
+            x = paddle.to_tensor(self.x)
+            y = paddle.to_tensor(self.y)
+            w = paddle.to_tensor(np.full(4, 0.5).astype(self.dtype))
+            out = paddle.lerp(x, y, w)
+            self.assertEqual(np.allclose(self.res_ref, out.numpy()), True)
+            paddle.enable_static()
+
+        for place in self.place:
+            run(place)
+
+    def test_x_broadcast_y(self):
+        paddle.disable_static()
+        x = np.arange(1., 21.).astype(self.dtype).reshape([2, 2, 5])
+        y = np.full(30, 10.).astype(self.dtype).reshape([3, 2, 1, 5])
+        out = paddle.lerp(paddle.to_tensor(x), paddle.to_tensor(y), 0.5)
+        res_ref = x + 0.5 * (y - x)
+        self.assertEqual(np.allclose(res_ref, out.numpy()), True)
+        paddle.enable_static()
+
+    def test_x_y_broadcast_w(self):
+        paddle.disable_static()
+        x = np.arange(1., 11.).astype(self.dtype).reshape([2, 5])
+        y = np.full(20, 10.).astype(self.dtype).reshape([2, 2, 5])
+        w = np.full(40, 0.5).astype(self.dtype).reshape([2, 2, 2, 5])
+        out = paddle.lerp(
+            paddle.to_tensor(x), paddle.to_tensor(y), paddle.to_tensor(w))
+        res_ref = x + w * (y - x)
```

Comment:
Can we add a test that the weight is not 0.5?

Reply:
done